src/main/java/com/project/Investment/App/service/impl/jdbc/EntityServiceJdbcImpl.java

```diff
@@ -18,46 +19,50 @@
 @Slf4j
 public class EntityServiceJdbcImpl implements EntityService {
 
-    private static final String URL = "jdbc:h2:mem:investment";
-    private static final String USERNAME = "sa";
-    private static final String PASSWORD = "";
+    private final Connection connection;
 
-    private static Connection connection;
-
-    static {
-        try {
-            Class.forName("org.h2.Driver");
-        } catch (ClassNotFoundException sqlException) {
-            sqlException.printStackTrace();
-        }
-
-        try {
-            connection = DriverManager.getConnection(URL, USERNAME, PASSWORD);
-        } catch (SQLException sqlException) {
-            sqlException.printStackTrace();
-        }
+    public EntityServiceJdbcImpl( Connection connection) {
+        this.connection = connection;
     }
 
     @Override
-    public Entity findById(String id) {
-        Entity entity = null;
+    public List<Entity> findById(String id, Optional<LocalDate> effectiveDate, Optional<Integer> limit) {
+        List<Entity> entity = new ArrayList<>();
         try {
+            ResultSet resultSet = null;
             PreparedStatement preparedStatement = connection.prepareStatement(QuerySQL.FIND_ENTITY_BY_ENTITY_ID_SQL);
-            preparedStatement.setString(1, id);
+            if (limit.isPresent()) {
```

Comment:
Try to make it more versatile: build query depending on available parameters using `MessageFormat`

Reply:
Done, but I had to use a ternary expression, because the effectiveDate in sql can be either the correct *' 2000-01-01 '* or the incorrect *' null '*, depending on whether it is present at the request. Also, there were thoughts to use *Querydsl / jooq*, but since not quite familiar with them refused.